kroxylicious-runtime/src/main/java/io/kroxylicious/proxy/internal/KafkaProxyInitializer.java

```diff
@@ -308,18 +311,20 @@ static class InitalizerNetFilter implements NetFilter {
         @Override
         public void selectServer(NetFilter.NetFilterContext context) {
             List<FilterAndInvoker> apiVersionFilters = decodePredicate.isAuthenticationOffloadEnabled() ? List.of()
-                    : FilterAndInvoker.build("ApiVersionsIntersect (internal)", apiVersionsIntersectFilter);
+                    : FilterAndInvoker.build(new FilterOptions("ApiVersionsIntersect (internal)", TargetMessageClass.ALL), apiVersionsIntersectFilter);
 
             NettyFilterContext filterContext = new NettyFilterContext(ch.eventLoop(), pfr);
             List<FilterAndInvoker> filterChain = filterChainFactory.createFilters(filterContext, filterDefinitions);
-            List<FilterAndInvoker> brokerAddressFilters = FilterAndInvoker.build("BrokerAddress (internal)", new BrokerAddressFilter(gateway, endpointReconciler));
+            List<FilterAndInvoker> brokerAddressFilters = FilterAndInvoker.build(new FilterOptions("BrokerAddress (internal)", TargetMessageClass.ALL),
+                    new BrokerAddressFilter(gateway, endpointReconciler));
             var filters = new ArrayList<>(apiVersionFilters);
-            filters.addAll(FilterAndInvoker.build("ApiVersionsDowngrade (internal)", apiVersionsDowngradeFilter));
+            filters.addAll(FilterAndInvoker.build(new FilterOptions("ApiVersionsDowngrade (internal)", TargetMessageClass.ALL), apiVersionsDowngradeFilter));
             filters.addAll(filterChain);
             if (binding.restrictUpstreamToMetadataDiscovery()) {
-                filters.addAll(FilterAndInvoker.build("EagerMetadataLearner (internal)", new EagerMetadataLearner()));
+                filters.addAll(FilterAndInvoker.build(new FilterOptions("EagerMetadataLearner (internal)", TargetMessageClass.ALL), new EagerMetadataLearner()));
             }
             filters.addAll(brokerAddressFilters);
+            filters.addAll(FilterAndInvoker.build(new FilterOptions("TopicNameLearner (internal)", TargetMessageClass.INTERNAL_EDGE_CACHEABLE), new TopicNameLearner()));
```

Comment:
relating this to the proposal, this is where we configure the filter to only intercept messages that are 'edge-cacheable'